src/openvic-simulation/interface/GFXObject.hpp

```diff
@@ -74,4 +77,232 @@ namespace OpenVic::GFX {
+	};
+
+	/* mapitems.gfx */
+	class Projection final : public Object {
+		friend std::unique_ptr<Projection> std::make_unique<Projection>();
+	
+	private:
+		std::string PROPERTY(texture_file);
+		//TODO: pulseSpeed, fadeout be ints or fixed points? assume fixed_point_t to start
+		fixed_point_t PROPERTY(size);
+		fixed_point_t PROPERTY(spin);
+		bool PROPERTY(pulsating);
+		fixed_point_t PROPERTY(pulse_lowest);
+		fixed_point_t PROPERTY(pulse_speed);
+		bool PROPERTY(additative);
+		fixed_point_t PROPERTY(expanding);
+		std::optional<fixed_point_t> PROPERTY(duration); //if present, determines how long until the projection disappears
+		std::optional<fixed_point_t> PROPERTY(fadeout); //appears to have no affect
+
+	protected:
+		Projection();
+
+		bool _fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) override;
+
+	public:
+		Projection(Projection&&) = default;
+		virtual ~Projection() = default;
+
+		OV_DETAIL_GET_TYPE
+	};
+
+	class Billboard final : public Object {
+		friend std::unique_ptr<Billboard> std::make_unique<Billboard>();
+
+	public:
+
+	private:
+		std::string PROPERTY(texture_file);
+		fixed_point_t PROPERTY(scale);
+		int64_t PROPERTY(no_of_frames);
+		int64_t PROPERTY(font_size); //assume font size is int32, not fixed_point
+		fvec3_t PROPERTY(offset); //TODO
+		std::string PROPERTY(font);
+
+	protected:
+		Billboard();
+
+		bool _fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) override;
+
+	public:
+		Billboard(Billboard&&) = default;
+		virtual ~Billboard() = default;
+
+		OV_DETAIL_GET_TYPE
+	};
+
+	class UnitStatsBillboard final : public Object {
+		friend std::unique_ptr<UnitStatsBillboard> std::make_unique<UnitStatsBillboard>();
+
+	private:
+		std::string PROPERTY(texture_file);
+		std::string PROPERTY(effect_file);
+		std::string PROPERTY(mask_file);
+		fixed_point_t PROPERTY(scale);
+		int64_t PROPERTY(no_of_frames);
+		int64_t PROPERTY(font_size); //assume font size is int, not fixed_point
+		std::string PROPERTY(font);
+
+	protected:
+		UnitStatsBillboard();
+
+		bool _fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) override;
+
+	public:
+		UnitStatsBillboard(UnitStatsBillboard&&) = default;
+		virtual ~UnitStatsBillboard() = default;
+
+		OV_DETAIL_GET_TYPE
+	};
+
+	class ProgressBar3d final : public Object {
+		friend std::unique_ptr<ProgressBar3d> std::make_unique<ProgressBar3d>();
+
+	private:
+		colour_t PROPERTY(back_colour);
+		colour_t PROPERTY(progress_colour);
+		ivec2_t PROPERTY(size);
+		std::string PROPERTY(effect_file);
+
+	protected:
+		ProgressBar3d();
+
+		bool _fill_key_map(NodeTools::case_insensitive_key_map_t& key_map) override;
+
+	public:
+		ProgressBar3d(ProgressBar3d&&) = default;
+		virtual ~ProgressBar3d() = default;
+
+		OV_DETAIL_GET_TYPE
+	};
+
+	/* Core.gfx */
+	class AnimatedMapText final : public Object {
+		friend std::unique_ptr<AnimatedMapText> std::make_unique<AnimatedMapText>();
+
+		using enum text_format_t;
+
+	public:
+		struct TextBlock {
```

Comment:
Unless there can be multiple text blocks, might as well just have its contents as regular `AnimatedMapText` variables without the intermediate struct